routing/missioncontrol_store.go

```diff
@@ -222,38 +234,116 @@ func deserializeResult(k, v []byte) (*paymentResult, error) {
 
 // AddResult adds a new result to the db.
 func (b *missionControlStore) AddResult(rp *paymentResult) error {
-	return kvdb.Update(b.db, func(tx kvdb.RwTx) error {
+	b.queueMx.Lock()
+	defer b.queueMx.Unlock()
+	b.queue.PushBack(rp)
```

Comment:
Hmm, so the "results" are never actually read from the store, only on startup? Pathfinding always uses its internal memory representation?

Reply:
Exactly. Also I was thinking maybe the whole design of the mission control store is wrong since we use the keys for ordering to be able to delete old ones. Was thinking about a circular list based design where we can deterministically overwrite old entries, so no deletion would be needed but didn't really expand on the idea yet.